src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -2527,6 +2531,40 @@ bool size_fits_all_mem_uses(AddPNode* addp, int shift) {
   return true;
 }
 
+// Convert BootTest condition to Assembler condition.
+// Replicate the logic of cmpOpOper::ccode() and cmpOpUOper::ccode().
+Assembler::Condition booltest_cond_to_assembler_cond(BoolTest::mask cond) {
+  switch(cond) {
+    case BoolTest::eq:
```

Comment:
I'd `assert(cmpOpOper(BoolTest::overflow).ccode() == result)` here.

Reply:
Thanks, I think this assert will guarantee the correctness of this function.

I found that cmpOpUOper convert signed BoolTest conditions (gt, ge, lt, le) to unsigned Assembler conditions, but unsigned vector comparison in vector API will produce unsigned BoolTest conditions(uge, ugt, ult, ule), and these conditons cannot be passed like following:

```
assert(cmpOpUOper(unsigned_cond).ccode() == result, "Invalid conversion");
``` 
Maybe we will meet some issues when taking unsigned vector comparison.

Reply:
I see. That sounds like a bug, but OK.

Reply:
Following is the code of ccode() of cmpOpUOper:

```
virtual int ccode() const { 
    switch (_c0) {
    case  BoolTest::eq : return equal();
    case  BoolTest::gt : return greater();
    case  BoolTest::lt : return less();
    case  BoolTest::ne : return not_equal();
    case  BoolTest::le : return less_equal();
    case  BoolTest::ge : return greater_equal();
    case  BoolTest::overflow : return overflow();
    case  BoolTest::no_overflow: return no_overflow();
    default : ShouldNotReachHere(); return 0;
    }
  };
```

I have another patch working on enabling SVE vector unsigned comparison, if we use ```assert(cmpOpUOper(unsigned_cond).ccode() == result, "Invalid conversion");```, the code will enter 
```ShouldNotReachHere()```.

Reply:
I see, but all you have to do then is something like

```
if (cond & BoolTest::unsigned_compare)
  assert( cmpOpUOper(cond & something).ccode() == result)
else
  assert( cmpOpOper(cond).ccode() == result)

```

surely?